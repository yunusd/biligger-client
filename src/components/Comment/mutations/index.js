import gql from 'graphql-tag';

const ADD_COMMENT = gql`
  mutation addComment(
    $content: String!,
    $parent: ID!,
    $parentModel: String!,
  ){
    addComment(content: $content, parent: $parent, parentModel: $parentModel){
      content
    }
  }
`;

const EDIT_COMMENT = gql`
  mutation editComment(
    $id: ID!,
    $content: String!,
    $author: ID!,
  ){
    editComment(id: $id, content: $content, author: $author){
      content
    }
  }
`;

const DELETE_COMMENT = gql`
  mutation deleteComment(
    $id: ID!,
    $author: ID!,
  ){
    deleteComment(id: $id, author: $author){
      content
    }
  }
`;

export {
  ADD_COMMENT,
  EDIT_COMMENT,
  DELETE_COMMENT,
};
