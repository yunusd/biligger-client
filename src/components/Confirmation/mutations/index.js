import gql from 'graphql-tag';

const RESET_USER_PASSWORD = gql`
  mutation resetUserPassword($newPassword: String!, $newPasswordCheck: String!, $hash: String!){
    resetUserPassword(newPassword: $newPassword, newPasswordCheck: $newPasswordCheck, hash: $hash){
      username
    }
  }
`;

const SEND_CONFIRMATION_EMAIL = gql`
  mutation sendConfirmationEmail($email: String!, $action: Int!){
    sendConfirmationEmail(email: $email, action: $action){
      hash,
      expired,
    }
  }
`;

export {
  RESET_USER_PASSWORD,
  SEND_CONFIRMATION_EMAIL,
};
