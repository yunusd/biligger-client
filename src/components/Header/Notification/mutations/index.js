import gql from 'graphql-tag';

const SEEN_NOTIFICATION = gql`
  mutation seenNotification {
    seenNotification {
      seen
    }
  }
`;

const DELETE_NOTIFICATION = gql`
  mutation deleteNotification {
    deleteNotification {
      seen
    }
  }
`;

export {
  SEEN_NOTIFICATION,
  DELETE_NOTIFICATION,
};
