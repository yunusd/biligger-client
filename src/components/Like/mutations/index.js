import gql from 'graphql-tag';

const ADD_LIKE = gql`
  mutation addLike($id: ID!, $parentModel: String!) {
    addLike(id: $id, parentModel: $parentModel) {
      liked
    }
  }
`;

const REMOVE_LIKE = gql`
  mutation removeLike($id: ID!, $parentModel: String!) {
    removeLike(id: $id, parentModel: $parentModel) {
      liked
    }
  }
`;

export {
  ADD_LIKE,
  REMOVE_LIKE,
};
