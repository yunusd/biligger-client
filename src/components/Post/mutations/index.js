import gql from 'graphql-tag';

const ADD_POST = gql`
  mutation addPost(
    $title: String!,
    $content: String!,
    $url: String,
    $category: String!,
  ){
    addPost(
      title: $title,
      content: $content,
      url:  $url,
      category: $category,
    ){
      id
      title
    }
  }
`;

const EDIT_POST = gql`
  mutation editPost(
    $id: ID!,
    $title: String!,
    $content: String!,
    $url: String,
    $category: String!,
    $author: ID!,
  ){
    editPost(
      id: $id,
      title: $title,
      content: $content,
      url:  $url,
      category: $category,
      author: $id,
    ){
      title
    }
  }
`;

export { ADD_POST, EDIT_POST };
