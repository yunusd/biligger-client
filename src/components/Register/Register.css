.modal-content {
  padding:0!important;
}

.register-left{
  background-color: #5B7EBB;
  width: 40%
}
.register-left .header{
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  color:white
}

.description form {
  padding: 20px
}

.description form .button {
  margin:0 0 10px 0;
  color: white;
  background-color: #5B7EBB;
}