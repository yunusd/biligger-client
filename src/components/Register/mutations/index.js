import gql from 'graphql-tag';

const REGISTER_USER = gql`
  mutation registerUser(
      $username: String!,
      $password: String!,
      $passwordCheck: String!,
      $email: String!,
      $degree: String,
      $bio: String,
    ){
    registerUser(
      username: $username,
      email: $email,
      password: $password,
      passwordCheck: $passwordCheck,
      degree: $degree,
      bio: $bio,
    ){
      username
    }
  }
`;

export {
  REGISTER_USER,
};
