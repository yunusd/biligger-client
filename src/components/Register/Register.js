import React, { useEffect } from 'react';
import { useApolloClient } from 'react-apollo-hooks';
import { Mutation } from 'react-apollo';
import { Link, Redirect } from 'react-router-dom';
import {
  Grid, Button, Header, Segment, Form, Divider, Image, Message,
} from 'semantic-ui-react';
import { withFormik } from 'formik';
import * as Yup from 'yup';

import { REGISTER_USER } from './mutations';
import { GET_AUTH_STATUS } from '../../queries';
import LogIn from '../LogIn';
import logo from '../../logo.png';
import './Register.css';

const RegisterSchema = Yup.object().shape({
  username: Yup.string()
    .min(3, 'Kullanıcı adı çok kısa!')
    .max(20, 'Kullanıcı adı çok uzun!')
    .required('Kullanıcı adı gerekli!'),
  password: Yup.string()
    .required('Şifre gerekli!')
    .matches(
      /^(?=(.*[a-zA-Z].*){3,})(?=.*\d.*)(?=.*\W.*)[a-zA-Z0-9\S]{8,}$/,
      'Şifre 8 karakter uzunluğunda olmalı ve en az 3 harf, 1 rakam ve 1 özel karakter içermelidir!',
    ),
  passwordCheck: Yup.string()
    .oneOf([Yup.ref('password'), null], 'Şifreler eşleşmeli')
    .required('Şifre(tekrar) gerekli!'),
  email: Yup.string()
    .email('E-Posta adresi geçersiz!')
    .required('E-Posta gerekli!'),
  degree: Yup.string()
    .min(3, 'Ünvan çok kısa!')
    .max(30, 'Ünvan çok uzun!')
    .required('Ünvan gerekli!'),
  bio: Yup.string()
    .min(30, 'Hakkında çok kısa!')
    .max(300, 'Hakkında çok uzun!'),
  invitationCode: Yup.string()
    .min(10, 'Davetiye kodu geçersiz!')
    .max(10, 'Davetiye kodu geçersiz!')
    .required('Üye olabilmeniz için davetiye kodu gerekli!'),

});

const RegisterForm = (props) => {
  const client = useApolloClient();

  const { currentUser } = client.readQuery({ query: GET_AUTH_STATUS });
  if (currentUser.isLoggedIn) return <Redirect to="/" />;

  const {
    values,
    touched,
    errors,
    handleChange,
    handleSubmit,
    status,
    isSubmitting,
  } = props;

  useEffect(() => {
    document.getElementById('root').style.background = 'linear-gradient(to top, #527ec0 68.3%, #ffffff 50%)';
    document.getElementById('root').style.position = 'absolute';
    document.getElementById('root').style.width = '100%';
    document.getElementById('root').style.height = '100%';

    return () => {
      document.getElementById('root').style.background = null;
      document.getElementById('root').style.position = null;
      document.getElementById('root').style.width = null;
      document.getElementById('root').style.height = null;
    };
  });

  return (
    <Mutation mutation={REGISTER_USER}>
      {(register, { loading }) => {
        values.register = register;

        const errorExists = Object.getOwnPropertyNames(errors).length > 0
        && Object.getOwnPropertyNames(touched).length > 0;
        const serverValidationErrors = status.errors ? status.errors : null;

        return (
          <div>
            {!status.isRegister
              ? (
                <div className="login-form">
                  <Grid textAlign="center" className="register-grid" verticalAlign="bottom">
                    <Grid.Column className="register-column">
                      <Image as={Link} to="/" src={logo} size="medium" />
                      <Header as="h2" className="register-header" textAlign="center">
                        Hoşgeldiniz
                      </Header>
                      <Segment raised className="register-segment">
                        {
                          errorExists || !!serverValidationErrors
                          ? (
                            <Message
                              error
                              list={[
                                serverValidationErrors && serverValidationErrors.username,
                                serverValidationErrors && serverValidationErrors.email,
                                serverValidationErrors && serverValidationErrors.invitationCode,
                                errors.invitationCode,
                                errors.username,
                                errors.email,
                                errors.password,
                                errors.passwordCheck,
                                errors.degree,
                                errors.bio,
                              ]}
                            />
                          )
                          : null
                        }
                        <Form
                          loading={loading}
                          onSubmit={(e) => {
                            e.preventDefault();
                            handleSubmit();
                          }}
                        >
                          <Form.Input error={!!errors.invitationCode || !!(serverValidationErrors && serverValidationErrors.invitationCode)} placeholder="Davetiye kodu" name="invitationCode" value={values.invitationCode} onChange={handleChange} />
                          <Form.Input error={!!errors.username || !!(serverValidationErrors && serverValidationErrors.username)} placeholder="Kullanıcı adı" name="username" value={values.username} onChange={handleChange} />
                          <Form.Input error={!!errors.email || !!(serverValidationErrors && serverValidationErrors.email)} type="email" placeholder="E-Posta adresi" name="email" value={values.email} onChange={handleChange} />
                          <Form.Input error={!!errors.password} type="password" placeholder="Şifre" name="password" value={values.password} onChange={handleChange} />
                          <Form.Input error={!!errors.passwordCheck} type="password" placeholder="Şifre(tekrar)" name="passwordCheck" value={values.passwordCheck} onChange={handleChange} />
                          <Form.Input error={!!errors.degree} placeholder="Ünvan" name="degree" value={values.degree} onChange={handleChange} />
                          <Form.TextArea error={!!errors.bio} placeholder="Hakkınızda" name="bio" value={values.bio} onChange={handleChange} />

                          <Button
                            disabled={isSubmitting}
                            type="submit"
                            content="Kayıt Ol"
                          />
                        </Form>
                        <Divider />
                        Zaten kayıtlı mısın?
                        <br />
                        <Link to="/giriş" style={{ color: '#49ba6f' }}>Giriş Yap</Link>
                      </Segment>
                    </Grid.Column>
                  </Grid>
                </div>
              )
              : <LogIn isRedirect username={status.username} />
            }
          </div>
        );
      }
    }
    </Mutation>
  );
};

const Register = withFormik({
  mapPropsToValues: () => ({
    invitationCode: '',
    username: '',
    email: '',
    password: '',
    passwordCheck: '',
    degree: '',
    bio: '',
  }),
  mapPropsToStatus: () => ({
    isRegister: false,
    errors: null,
  }),
  validationSchema: RegisterSchema,
  handleSubmit: async (values, { setSubmitting, setStatus }) => {
    const {
      register,
      invitationCode,
      username,
      password,
      passwordCheck,
      email,
      degree,
      bio,
    } = values;
    // Try catch block needed for testing. Without it test not passing.
    try {
      await register({
        variables: {
          invitationCode,
          username,
          password,
          passwordCheck,
          email,
          degree,
          bio,
        },
      });
      setStatus({ isRegister: true, username });
    } catch (err) {
      setSubmitting(false);
      const { errors, invalidArg } = err.graphQLErrors[0].extensions.exception;

      setStatus({
        errors: {
          username: errors && errors.username !== undefined && errors.username.message,
          email: errors && errors.email !== undefined && errors.email.message,
          invitationCode: invalidArg && err.graphQLErrors[0].message,
        },
      });
      return err;
    }
  },
  displayName: 'BasicForm',
})(RegisterForm);

export default Register;
