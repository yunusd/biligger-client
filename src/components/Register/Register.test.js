import React from 'react';
import { shallow, mount } from 'enzyme';
import { MockedProvider } from 'react-apollo/test-utils';

import {
  Button, Modal, Form, Header,
} from 'semantic-ui-react';

import { REGISTER_USER } from './mutations';
import Register from './Register';
import waita from '../../helpers/waita';

describe('Register', () => {
  const user = {
    username: 'foo',
    password: 'bar',
    passwordCheck: 'bar',
    email: 'foo@bar.com',
    degree: 'degg',
    bio: `
    This is a bio sentence and will repeat itself.
    This is a bio sentence and will repeat itself.
    This is a bio sentence and will repeat itself.
    This is a bio sentence and will repeat itself.
    `,
  };

  const {
    username,
    email,
    password,
    passwordCheck,
    degree,
    bio,
  } = user;

  const mocks = [
    {
      request: {
        query: REGISTER_USER,
        variables: {
          username,
          email,
          password,
          passwordCheck,
          degree,
          bio,
        },
      },
      result: { data: { registerUser: { username } } },
    },
  ];

  it('render without crashing', () => {
    const wrapper = shallow(<Register />).exists();
    expect(wrapper).toBeTruthy();
  });

  it('render correctly', () => {
    const wrapper = shallow(<Register />);
    expect(wrapper).toMatchSnapshot();
  });

  it('call handleChange function and set state by values', () => {
    const wrapper = mount(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Register button={<Button basic color="green">Kayıt Ol</Button>} />
      </MockedProvider>,
    );

    wrapper.find(Modal).find(Button).simulate('click');
    wrapper.find(Form.Input).at(0)
      .prop('onChange')({ target: { name: 'username', value: username } });

    wrapper.find(Form.Input).at(1)
      .prop('onChange')({ target: { name: 'email', value: email } });

    wrapper.find(Form.Input).at(2)
      .prop('onChange')({ target: { name: 'password', value: password } });

    wrapper.find(Form.Input).at(3)
      .prop('onChange')({ target: { name: 'passwordCheck', value: passwordCheck } });

    wrapper.find(Form.Input).at(4)
      .prop('onChange')({ target: { name: 'degree', value: degree } });

    wrapper.find(Form.TextArea).at(0)
      .prop('onChange')({ target: { name: 'bio', value: bio } });

    const state = wrapper.find(Register).state();
    expect(state).toMatchObject(user);
  });

  it('if credentials are valid then register user', async () => {
    const wrapper = mount(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Register button={<Button basic color="green">Kayıt Ol</Button>} />
      </MockedProvider>,
    );
    wrapper.find(Modal).find(Button).simulate('click');
    wrapper.find(Register).setState(user);
    wrapper.find(Form).find(Button).simulate('submit');
    await waita(2);
    const logInModal = wrapper.update().find(Modal).find(Header);
    expect(logInModal.text()).toContain(username);
  });

  /*
    Apollo waiting a variables which match with mock variables.
    Apollo will throw an error because we don't send user credentials(Variables) and this is
    causing to test not passing. So we used try catch block on handleSubmit function
    because of without it, test not passing.
    But this time it's allowing test to run.
    Component will render error message.
  */
  it('if credentials are wrong then return an error', async () => {
    const wrapper = mount(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Register button={<Button basic color="green">Kayıt Ol</Button>} />
      </MockedProvider>,
    );
    // We don't sending any information
    wrapper.find(Modal).find(Button).simulate('click');
    wrapper.find(Form).find(Button).simulate('submit');
    await waita(2);
    expect(wrapper.find(Header).text()).toContain('Hata');
  });
});
