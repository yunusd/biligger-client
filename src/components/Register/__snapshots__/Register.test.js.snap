// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Register />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
      "mutation": Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": Object {
              "kind": "Name",
              "value": "registerUser",
            },
            "operation": "mutation",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "username",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "username",
                        },
                      },
                    },
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "email",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "email",
                        },
                      },
                    },
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "password",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "password",
                        },
                      },
                    },
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "passwordCheck",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "passwordCheck",
                        },
                      },
                    },
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "degree",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "degree",
                        },
                      },
                    },
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "bio",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "bio",
                        },
                      },
                    },
                  ],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "registerUser",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "username",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                },
              ],
            },
            "variableDefinitions": Array [
              Object {
                "defaultValue": undefined,
                "kind": "VariableDefinition",
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "String",
                    },
                  },
                },
                "variable": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "username",
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "kind": "VariableDefinition",
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "String",
                    },
                  },
                },
                "variable": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "password",
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "kind": "VariableDefinition",
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "String",
                    },
                  },
                },
                "variable": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "passwordCheck",
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "kind": "VariableDefinition",
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "String",
                    },
                  },
                },
                "variable": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "email",
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "kind": "VariableDefinition",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "String",
                  },
                },
                "variable": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "degree",
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "kind": "VariableDefinition",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "String",
                  },
                },
                "variable": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "bio",
                  },
                },
              },
            ],
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 384,
          "start": 0,
        },
      },
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
        "mutation": Object {
          "definitions": Array [
            Object {
              "directives": Array [],
              "kind": "OperationDefinition",
              "name": Object {
                "kind": "Name",
                "value": "registerUser",
              },
              "operation": "mutation",
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "alias": undefined,
                    "arguments": Array [
                      Object {
                        "kind": "Argument",
                        "name": Object {
                          "kind": "Name",
                          "value": "username",
                        },
                        "value": Object {
                          "kind": "Variable",
                          "name": Object {
                            "kind": "Name",
                            "value": "username",
                          },
                        },
                      },
                      Object {
                        "kind": "Argument",
                        "name": Object {
                          "kind": "Name",
                          "value": "email",
                        },
                        "value": Object {
                          "kind": "Variable",
                          "name": Object {
                            "kind": "Name",
                            "value": "email",
                          },
                        },
                      },
                      Object {
                        "kind": "Argument",
                        "name": Object {
                          "kind": "Name",
                          "value": "password",
                        },
                        "value": Object {
                          "kind": "Variable",
                          "name": Object {
                            "kind": "Name",
                            "value": "password",
                          },
                        },
                      },
                      Object {
                        "kind": "Argument",
                        "name": Object {
                          "kind": "Name",
                          "value": "passwordCheck",
                        },
                        "value": Object {
                          "kind": "Variable",
                          "name": Object {
                            "kind": "Name",
                            "value": "passwordCheck",
                          },
                        },
                      },
                      Object {
                        "kind": "Argument",
                        "name": Object {
                          "kind": "Name",
                          "value": "degree",
                        },
                        "value": Object {
                          "kind": "Variable",
                          "name": Object {
                            "kind": "Name",
                            "value": "degree",
                          },
                        },
                      },
                      Object {
                        "kind": "Argument",
                        "name": Object {
                          "kind": "Name",
                          "value": "bio",
                        },
                        "value": Object {
                          "kind": "Variable",
                          "name": Object {
                            "kind": "Name",
                            "value": "bio",
                          },
                        },
                      },
                    ],
                    "directives": Array [],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "registerUser",
                    },
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "username",
                          },
                          "selectionSet": undefined,
                        },
                      ],
                    },
                  },
                ],
              },
              "variableDefinitions": Array [
                Object {
                  "defaultValue": undefined,
                  "kind": "VariableDefinition",
                  "type": Object {
                    "kind": "NonNullType",
                    "type": Object {
                      "kind": "NamedType",
                      "name": Object {
                        "kind": "Name",
                        "value": "String",
                      },
                    },
                  },
                  "variable": Object {
                    "kind": "Variable",
                    "name": Object {
                      "kind": "Name",
                      "value": "username",
                    },
                  },
                },
                Object {
                  "defaultValue": undefined,
                  "kind": "VariableDefinition",
                  "type": Object {
                    "kind": "NonNullType",
                    "type": Object {
                      "kind": "NamedType",
                      "name": Object {
                        "kind": "Name",
                        "value": "String",
                      },
                    },
                  },
                  "variable": Object {
                    "kind": "Variable",
                    "name": Object {
                      "kind": "Name",
                      "value": "password",
                    },
                  },
                },
                Object {
                  "defaultValue": undefined,
                  "kind": "VariableDefinition",
                  "type": Object {
                    "kind": "NonNullType",
                    "type": Object {
                      "kind": "NamedType",
                      "name": Object {
                        "kind": "Name",
                        "value": "String",
                      },
                    },
                  },
                  "variable": Object {
                    "kind": "Variable",
                    "name": Object {
                      "kind": "Name",
                      "value": "passwordCheck",
                    },
                  },
                },
                Object {
                  "defaultValue": undefined,
                  "kind": "VariableDefinition",
                  "type": Object {
                    "kind": "NonNullType",
                    "type": Object {
                      "kind": "NamedType",
                      "name": Object {
                        "kind": "Name",
                        "value": "String",
                      },
                    },
                  },
                  "variable": Object {
                    "kind": "Variable",
                    "name": Object {
                      "kind": "Name",
                      "value": "email",
                    },
                  },
                },
                Object {
                  "defaultValue": undefined,
                  "kind": "VariableDefinition",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "String",
                    },
                  },
                  "variable": Object {
                    "kind": "Variable",
                    "name": Object {
                      "kind": "Name",
                      "value": "degree",
                    },
                  },
                },
                Object {
                  "defaultValue": undefined,
                  "kind": "VariableDefinition",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "String",
                    },
                  },
                  "variable": Object {
                    "kind": "Variable",
                    "name": Object {
                      "kind": "Name",
                      "value": "bio",
                    },
                  },
                },
              ],
            },
          ],
          "kind": "Document",
          "loc": Object {
            "end": 384,
            "start": 0,
          },
        },
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
